pkg/controllers/startup_policy_test.go

```diff
@@ -0,0 +1,159 @@
+		startupPolicy *jobset.StartupPolicy
+		expected      bool
+	}{
+		{
+			name:          "startup policy AnyOrder",
+			startupPolicy: &jobset.StartupPolicy{StartupPolicyOrder: jobset.AnyOrder},
+			expected:      false,
+		},
+		{
+			name:          "startup policy InOrder",
+			startupPolicy: &jobset.StartupPolicy{StartupPolicyOrder: jobset.InOrder},
+			expected:      true,
+		},
+		{
+			name:          "nil startup policy",
+			startupPolicy: nil,
+			expected:      false,
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			actual := inOrderStartupPolicy(tc.startupPolicy)
+			if diff := cmp.Diff(tc.expected, actual); diff != "" {
+				t.Errorf("unexpected finished value (+got/-want): %s", diff)
+			}
+		})
+	}
+}
+
+func TestReplicatedJobStarted(t *testing.T) {
+	tests := []struct {
+		name                string
+		replicas            int32
+		replicatedJobStatus jobset.ReplicatedJobStatus
+		expected            bool
+	}{
+		{
+			name:                "replicas 1; no replicatedJobStatus",
+			replicas:            1,
+			replicatedJobStatus: jobset.ReplicatedJobStatus{},
+			expected:            false,
+		},
+		{
+			name:     "replicas 4; replicatedJobStatus all ready",
+			replicas: 4,
+			replicatedJobStatus: jobset.ReplicatedJobStatus{
+				Name:      "test",
+				Ready:     4,
+				Succeeded: 0,
+				Failed:    0,
+				Suspended: 0,
+				Active:    0,
+			},
+			expected: true,
+		},
+		{
+			name:     "replicas 4; replicatedJobStatus mix of ready, failed and succeded",
+			replicas: 4,
+			replicatedJobStatus: jobset.ReplicatedJobStatus{
+				Name:      "test",
+				Ready:     2,
+				Succeeded: 1,
+				Failed:    1,
+				Suspended: 0,
+				Active:    0,
+			},
+			expected: true,
+		},
+
+		{
+			name:     "replicas 4; replicatedJobStatus all active",
+			replicas: 4,
+			replicatedJobStatus: jobset.ReplicatedJobStatus{
+				Name:      "test",
+				Ready:     0,
+				Succeeded: 0,
+				Failed:    0,
+				Suspended: 0,
+				Active:    4,
+			},
+			expected: false,
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			actual := replicatedJobsStarted(tc.replicas, tc.replicatedJobStatus)
+			if diff := cmp.Diff(tc.expected, actual); diff != "" {
+				t.Errorf("unexpected finished value (+got/-want): %s", diff)
+			}
+		})
+	}
+}
+
+func TestGenerateStartupPolicyCondition(t *testing.T) {
+	tests := []struct {
+		name              string
+		policyComplete    bool
+		jobName           string
+		expectedCondition metav1.Condition
+	}{
+		{
+			name:           "in progress startup condition on a",
+			policyComplete: false,
+			jobName:        "a",
+			expectedCondition: metav1.Condition{
+				Type:    string(jobset.JobSetStartupPolicyCompleted),
+				Status:  metav1.ConditionFalse,
+				Reason:  "StartupPolicyInOrder",
+				Message: "startup policy in order starting",
+			},
+		},
+		{
+			name:           "startup policy complete",
+			policyComplete: true,
+			jobName:        "",
```

Comment:
why are we setting jobName as an empty string here?

Reply:
Nice find. That wasn't actually used anymore so I removed it.